### PR TITLE
[full-ci] feat(smoke tests): automated parallel smoke tests

### PR DESCRIPTION
**Pull request by fschade**
This PR introduces a first step to have automated smoke tests.

To get a better understanding what that is, what not and how this could be useful pleas read the [README.md](https://github.com/owncloud/web/blob/add-smoke-tests/tests/smoke/README.md)

It introduces new concepts how and where tests are running.
It's not a final solutions, instead it's just a starting point and needs to be extended if agreed to go that road.

Things that not there yet:
* Persona does not include all smoke tests
* runs only local, Drone setup is yet not there

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oC10MoveUpload</strong> https://drone.owncloud.com/owncloud/web/18295/14/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10iPhone1</strong> https://drone.owncloud.com/owncloud/web/18296/45/1
The following scenarios passed on retry:
- webUIDeleteFilesFolders/deleteFilesFolders.feature:59

**Comment by ownclouders** [comment]
Results for <strong>oC10XGAPortrait2</strong> https://drone.owncloud.com/owncloud/web/18296/43/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by phil-davis** [comment]
I see that this uses Playwright as the framework for accessing the browser. That moves away from Nightwatch.

We already started making some user-journeys. They are currently in the Nightwatch-based test suite.

It is easy to add whatever scenarios we want to the Nightwatch-based test suites. We can add to the user-journeys, or we can make another test suite for "basic smoke tests for local development" or we can add and choose scenarios anywhere through the test suites and put some tag on them so that they can be easily run.

There is a lot of duplication in having both Nightwatch and Playwright running scenarios that require the same knowledge of how to create a user, group, of the element ids on pages and so on.

What are the thoughts/reasons for moving from Nightwatch to Playwright?

If Playwright has significant advantage compared to Nightwatch, then should we switch all the UI tests to Playwright?

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationApp2</strong> https://drone.owncloud.com/owncloud/web/18319/66/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by fschade** [comment]
> I see that this uses Playwright as the framework for accessing the browser. That moves away from Nightwatch.
> 
> We already started making some user-journeys. They are currently in the Nightwatch-based test suite.
> 
> It is easy to add whatever scenarios we want to the Nightwatch-based test suites. We can add to the user-journeys, or we can make another test suite for "basic smoke tests for local development" or we can add and choose scenarios anywhere through the test suites and put some tag on them so that they can be easily run.
> 
> There is a lot of duplication in having both Nightwatch and Playwright running scenarios that require the same knowledge of how to create a user, group, of the element ids on pages and so on.
> 
> What are the thoughts/reasons for moving from Nightwatch to Playwright?
> 
> If Playwright has significant advantage compared to Nightwatch, then should we switch all the UI tests to Playwright?

Thanks for you input, it started to be a poc and at the end worked out really nice.
Playwright has a totally different concept then selenium of how it is automating the browser.

This has a lot of benefits, for example there are rare cases where a manual wait is needed.
I see this as a POC which eventually can evolve over time. It should not test everything like we do in our mature acceptance test. Instead it's a tiny piece which fills the gap of quick, local testing before the code hits the ci.

**Comment by ownclouders** [comment]
Results for <strong>oC10XGAPortrait1</strong> https://drone.owncloud.com/owncloud/web/18463/42/1
The following scenarios passed on retry:
- webUIDeleteFilesFolders/deleteFilesFolders.feature:59

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingInternalUsers1</strong> https://drone.owncloud.com/owncloud/web/18463/55/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISLocks</strong> https://drone.owncloud.com/owncloud/web/18796/48/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic</strong> https://drone.owncloud.com/owncloud/web/18796/61/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10iPhone2</strong> https://drone.owncloud.com/owncloud/web/18806/46/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by pascalwengerter** [comment]
Rebased and fixed a minor CI issue, in favor of merging this and building on top/updating the (currently WIP/draft documentation)

File: tests/smoke/support/api/http.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        ...(user && {
+          Authorization: 'Basic ' + Buffer.from(user.id + ':' + user.password).toString('base64')
+        })
+      }
+    }
+  )
+}
+
+export const checkResponseStatus = (response: Response, message = ''): void => {
+  // response.status >= 200 && response.status < 300
```
Comment (pascalwengerter) [inline tests/smoke/support/api/http.ts:39]: Dev leftover?

File: tests/smoke/support/api/http.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import join from 'join-path'
+import fetch, { BodyInit, Response } from 'node-fetch'
+import { User } from '../types'
+import { config } from '../config'
+import _ from 'lodash'
```
Comment (pascalwengerter) [inline tests/smoke/support/api/http.ts:5]: Why not use `lodash-es` here?

**Review by pascalwengerter — APPROVED** [review]
Lovely!

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/web/18812/7/8
```
error owncloud-js-dev@0.0.1: The engine "node" is incompatible with this module. Expected version ">= 14.17.0". Got "8.17.0"
```

That is surprising - this branch is rebased to the latest master.

**Comment by pascalwengerter** [comment]
> https://drone.owncloud.com/owncloud/web/18812/7/8
> 
> ```
> error owncloud-js-dev@0.0.1: The engine "node" is incompatible with this module. Expected version ">= 14.17.0". Got "8.17.0"
> ```
> 
> That is surprising - this branch is rebased to the latest master.

From my understanding it lacks the node14-based core img, see latest commit 😉  and error msg in CI https://drone.owncloud.com/owncloud/web/18785/14/9

**Comment by phil-davis** [comment]
@pascalwengerter I updated my post while you were making yours.

This needs changes like in core PR https://github.com/owncloud/core/pull/39141

Should we do a separate PR here in web for that, and not mix it up with this, or?

**Comment by pascalwengerter** [comment]
> @pascalwengerter I updated my post while you were making yours.
> 
> This needs changes like in core PR [owncloud/core#39141](https://github.com/owncloud/core/pull/39141)
> 
> Should we do a separate PR here in web for that, and not mix it up with this, or?

This is where I got my inspiration from! Let's see how CI goes, I'm highly in favor of merging this if it gets green to not waste even more time waiting for it to pass

File: tests/smoke/support/api/http.ts (hunk 2 above)
Comment (fschade) [reply tests/smoke/support/api/http.ts:5]: we could but there it doesnt matter, does not get packed

File: tests/smoke/support/api/http.ts (hunk 1 above)
Comment (fschade) [reply tests/smoke/support/api/http.ts:39]: think yes, should be fine to get rid of it

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPublicManagement</strong> https://drone.owncloud.com/owncloud/web/18815/34/1
The following scenarios passed on retry:
- webUISharingPublicManagement/publicLinkIndicator.feature:27

**Pull request merged**